### PR TITLE
feat: alternative withdrawal address

### PR DESCRIPTION
**Pull request by notanatol**
### Checklist

- [X] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md).

### Description
Closes: #3676

### CODE REVIEW

File: pkg/api/wallet.go [hunk 1]
```diff
@@ -47,3 +54,101 @@ func (s *Service) walletHandler(w http.ResponseWriter, r *http.Request) {
 		WalletAddress:             s.ethereumAddress,
 	})
 }
+
+type walletTxResponse struct {
+	TransactionHash common.Hash `json:"transactionHash"`
+}
+
+func (s *Service) walletWithdrawHandler(w http.ResponseWriter, r *http.Request) {
```
Comment (istae) [inline pkg/api/wallet.go:62]: the new endpoint should be unit tested
the new integration tested is much appreciated, btw

Comment (notanatol) [reply pkg/api/wallet.go:62]: added

File: openapi/SwarmDebug.yaml [hunk 2]
```diff
@@ -1133,3 +1168,1160 @@ paths:
           $ref: "SwarmCommon.yaml#/components/responses/400"
         default:
           description: Default response.
+
+components:
```
Comment (istae) [inline openapi/SwarmDebug.yaml:1172]: why are so these components copied here from SwarmCommon.yaml?

Comment (notanatol) [reply openapi/SwarmDebug.yaml]: accidental copypaste, reverted

File: pkg/api/wallet.go [hunk 3]
```diff
@@ -47,3 +54,101 @@ func (s *Service) walletHandler(w http.ResponseWriter, r *http.Request) {
...
+		response("invalid query params", logger, w)
+		return
+	}
+
+	var bzz bool
+
+	if strings.EqualFold("BZZ", *path.Coin) {
+		bzz = true
+	} else if !strings.EqualFold("NativeToken", *path.Coin) {
+		logger.Error(nil, "invalid coin type")
```
Comment (acha-bill) [inline pkg/api/wallet.go:89]: This log is not necessary

File: pkg/api/wallet.go [hunk 4]
```diff
@@ -47,3 +54,101 @@ func (s *Service) walletHandler(w http.ResponseWriter, r *http.Request) {
...
+	nativeToken, err := s.chainBackend.BalanceAt(r.Context(), s.ethereumAddress, nil)
+	if err != nil {
+		logger.Debug("unable to acquire balance from the chain backend", "error", err)
+		logger.Error(nil, "unable to acquire balance from the chain backend")
+		jsonhttp.InternalServerError(w, "unable to acquire balance from the chain backend")
+		return
+	}
+
+	if queries.Amount.Cmp(nativeToken) > 0 {
+		logger.Error(err, "not enough balance")
```
Comment (acha-bill) [inline pkg/api/wallet.go:133]: This log is not necessary

**Review by istae — APPROVED** [review]

**Comment by darkobas2** [comment]
just a thought about this. I noticed it uses gasPrice for transaction which is deemed as Legacy transaction and not EIP-1559 maxPriorityFeePerGas and maxFeePerGas which is supposed to be recommended way by now and would probably prevent a failed transaction or two in case of network congestion.

**Pull request merged**